dotnet/src/Connectors/Connectors.Memory.DuckDB/Database.cs

```diff
@@ -145,15 +134,16 @@ private static string EncodeFloatArrayToString(float[]? data)
                 continue;
             }
 
-            string metadata = dataReader.GetString("metadata");
-            string embeddingAsString = dataReader.GetString("embeddingAsString");
-            string timestamp = dataReader.GetString("timestamp");
-            float score = dataReader.GetFloat("score");
+            string metadata = dataReader.IsDBNull("metadata") ? string.Empty : dataReader.GetString("metadata");
+            float[] embeddingFromSearch = dataReader.IsDBNull("embedding") ? Array.Empty<float>() : ((dataReader.GetValue("embedding") as List<float>)?.ToArray()) ?? Array.Empty<float>();
+            string timestamp = dataReader.IsDBNull("timestamp") ? string.Empty : dataReader.GetString("timestamp");
+            float score = dataReader.IsDBNull("score") ? 0f : dataReader.GetFloat("score");
```

Comment:
For primitive types, you can call `dataReader.GetFieldValue<float?>(2)` and you will get back `null` if the field is null.

Reply:
perfect, this is a great improvement!

Reply:
<img width="385" alt="image" src="https://github.com/microsoft/semantic-kernel/assets/375556/c9e33fc4-eaec-409f-a588-5dd521977727">

Reply:
<img width="352" alt="image" src="https://github.com/microsoft/semantic-kernel/assets/375556/3a88d260-d25a-42ee-b48b-cdfb6d33ac05">

Reply:
I think this image is for the previous comment.